elephant/statistics.py

```diff
@@ -799,23 +809,416 @@ def complexity_pdf(spiketrains, binsize):
+           data," In "Dynamic Brain - from Neural Spikes to Behaviors",
+           pp. 96-114, Springer Berlin Heidelberg, 2008.
+
+    Examples
+    --------
+    Here the behavior of
+    `elephant.spike_train_processing.precise_complexity_intervals` is shown, by
+    applying the function to some sample spiketrains.
+
+    >>> import neo
+    >>> import quantities as pq
+    >>> from elephant.statistics import complexity
+
+    >>> sr = 1/pq.ms
+
+    >>> st1 = neo.SpikeTrain([1, 4, 6] * pq.ms, t_stop=10.0 * pq.ms)
+    >>> st2 = neo.SpikeTrain([1, 5, 8] * pq.ms, t_stop=10.0 * pq.ms)
+    >>> sts = [st1, st2]
+
+    >>> # spread = 0, a simple bincount
+    >>> cpx = complexity(sts, sampling_rate=sr)
+    Complexity calculated at sampling rate precision
+    >>> print(cpx.histogram)
+    [5 4 1]
+    >>> print(cpx.time_histogram.flatten())
+    [0 2 0 0 1 1 1 0 1 0] dimensionless
+    >>> print(cpx.time_histogram.times)
+    [0. 1. 2. 3. 4. 5. 6. 7. 8. 9.] ms
+
+    >>> # spread = 1, consecutive spikes
+    >>> cpx = complexity(sts, sampling_rate=sr, spread=1)
+    Complexity calculated at sampling rate precision
+    >>> print(cpx.histogram)
+    [5 4 1]
+    >>> print(cpx.time_histogram.flatten())
+    [0 2 0 0 3 3 3 0 1 0] dimensionless
+
+    >>> # spread = 2, consecutive spikes and separated by 1 empty bin
+    >>> cpx = complexity(sts, sampling_rate=sr, spread=2)
+    Complexity calculated at sampling rate precision
+    >>> print(cpx.histogram)
+    [4 0 1 0 1]
+    >>> print(cpx.time_histogram.flatten())
+    [0 2 0 0 4 4 4 4 4 0] dimensionless
+    """
+
+    def __init__(self, spiketrains,
+                 sampling_rate=None,
+                 bin_size=None,
+                 binary=True,
+                 spread=0,
+                 tolerance=1e-8):
+
+        _check_consistency_of_spiketrainlist(spiketrains,
+                                             same_t_start=True,
+                                             same_t_stop=True)
+
+        if bin_size is None and sampling_rate is None:
+            raise ValueError('No bin_size or sampling_rate was specified!')
+
+        if spread < 0:
+            raise ValueError('Spread must be >=0')
+
+        self.input_spiketrains = spiketrains
+        self.t_start = spiketrains[0].t_start
+        self.t_stop = spiketrains[0].t_stop
+        self.sampling_rate = sampling_rate
+        self.bin_size = bin_size
+        self.binary = binary
+        self.spread = spread
+        self.tolerance = tolerance
+
+        if bin_size is None and sampling_rate is not None:
+            self.bin_size = 1 / self.sampling_rate
+
+        if spread == 0:
+            self.time_histogram, self.histogram = self._histogram_no_spread()
+            self.epoch = self._epoch_no_spread()
+        else:
+            self.epoch = self._epoch_with_spread()
+            self.time_histogram, self.histogram = self._histogram_with_spread()
+
+    @property
```

Comment:
Why property?
It is not a _characteristic_ of an object. PDF should be a function.